addons/stock/models/stock_picking.py

```diff
@@ -1163,6 +1163,22 @@ def button_validate(self):
             }
         return True
 
+    def action_split_transfer(self):
+        for picking in self:
+            if all(float_is_zero(m.quantity, precision_rounding=m.product_uom.rounding) for m in picking.move_ids):
+                raise UserError(_("%s: Nothing to split. Fill the quantities you want in a new transfer in the done quantities", picking.display_name))
+            if all(float_compare(m.quantity, m.product_uom_qty, precision_rounding=m.product_uom.rounding) == 0 for m in picking.move_ids):
+                raise UserError(_("%s: Nothing to split, all demand is done. For split you need at least one line not fully fulfilled", picking.display_name))
+            if any(float_compare(m.quantity, m.product_uom_qty, precision_rounding=m.product_uom.rounding) > 0 for m in picking.move_ids):
+                raise UserError(_("%s: Can't split: quantities done can't be above demand", picking.display_name))
+
+            moves = picking.move_ids.filtered(lambda m: m.state not in ('done', 'cancel'))
+            backorder_moves = moves._create_backorder()
+            moves_to_unlink = self.move_ids.filtered(lambda m: m.product_uom_qty == 0)
+            moves_to_unlink._action_cancel()
+            moves_to_unlink.unlink()
+            picking._create_backorder(backorder_moves=backorder_moves)
```

Comment:
remove from the loop `for picking in self` everything that can be done in batch

Reply:
It would be nice to keep empty moves and move them to backorder instead of delete them and recreate them